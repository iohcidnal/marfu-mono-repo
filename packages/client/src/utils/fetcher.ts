interface FetchConfig {
  url: string;
  method?: 'GET' | 'POST' | 'PUT' | 'DELETE';
  payload?: any;
}

interface FetchResult {
  status: number;
  statusText: string;
  data: any;
}

export default async function fetcher(
  config: FetchConfig = { url: null, method: 'GET', payload: null },
  requestContext = null
): Promise<FetchResult> {
  const option: RequestInit = {
    method: config.method,
    mode: 'cors',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/json'
    },
    ...(config.payload && { body: JSON.stringify(config.payload) })
  };

  if (requestContext) {
    // Use [] notation to make TS happy.
    option.headers['cookie'] = requestContext.req.headers.cookie;
  }

  console.log('option :>> ', option);
  const response: Response = await fetch(config.url, option);
  console.log('response :>> ', response);

  const data = await response.json();

  return {
    status: response.status,
    statusText: response.statusText,
    data
  };
}
