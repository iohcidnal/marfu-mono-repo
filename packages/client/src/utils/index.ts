export { default as fetcher } from './fetcher';
export type { IFetchResult } from './fetcher';
export { default as renderWithProviders } from './test-utils';
