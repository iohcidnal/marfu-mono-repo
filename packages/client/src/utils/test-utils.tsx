import * as React from 'react';
import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { QueryClientProvider, QueryClient } from 'react-query';

const queryClient = new QueryClient();

export default function renderWithProviders(children: React.ReactNode) {
  const Component = () => (
    <QueryClientProvider client={queryClient}>
      <ChakraProvider>{children}</ChakraProvider>
    </QueryClientProvider>
  );

  return () => {
    const rendered = render(<Component />);
    return {
      ...rendered
    };
  };
}
