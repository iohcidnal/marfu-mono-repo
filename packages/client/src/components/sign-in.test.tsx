import { act } from '@testing-library/react';
import { renderWithProviders } from '../utils';
import SignIn from './sign-in';

const render = renderWithProviders(<SignIn />);

describe('SignIn', () => {
  it('should render properly', async () => {
    let rendered;
    await act(async () => {
      rendered = render();
    });
    expect(rendered.asFragment()).toMatchSnapshot();
  });

  it.todo('cover other lines');
});
