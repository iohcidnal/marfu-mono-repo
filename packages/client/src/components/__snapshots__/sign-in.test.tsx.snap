// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-lgt9ky"
  >
    <div
      class="css-lr4539"
    >
      <h2
        class="chakra-heading css-14umdt2"
      >
        marfu
      </h2>
      <p
        class="chakra-text css-i9bggq"
      >
        Medication Administration Record for Family Use
      </p>
    </div>
    <div
      class="chakra-stack css-owjkmg"
    >
      <div
        class="chakra-stack css-7ek7do"
      >
        <div
          class="css-tnk16i"
        >
          <p
            class="chakra-text css-9dl0t2"
          >
            Sign in to continue
          </p>
          <form
            novalidate=""
          >
            <div
              class="chakra-stack css-b5leqj"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <label
                  class="chakra-form__label css-10m1y3x"
                  for="userName"
                  id="userName-label"
                >
                  Username
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-1ssjhh"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  class="chakra-input css-1jhp4gn"
                  id="userName"
                  name="userName"
                  required=""
                  type="email"
                />
              </div>
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="chakra-stack css-1yqpede"
                >
                  <label
                    class="chakra-form__label css-10m1y3x"
                    for="password"
                    id="password-label"
                  >
                    Password
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-1ssjhh"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <a
                    class="chakra-link css-34lw4l"
                  >
                    Forgot password?
                  </a>
                </div>
                <input
                  aria-required="true"
                  class="chakra-input css-1jhp4gn"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                />
              </div>
              <button
                class="chakra-button css-jg4a2p"
                disabled=""
                type="submit"
              >
                Sign in
              </button>
              <div
                class="chakra-stack css-mzzrj0"
              >
                <p
                  class="chakra-text css-0"
                >
                  New user?
                </p>
                <a
                  class="chakra-link css-t9eb1q"
                >
                  Create an account
                </a>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
